.github/DESIGN.md

```diff
@@ -0,0 +1,115 @@
+# Shopify App CLI design guidelines
+
+The purpose of this doc is to outline all the heuristics, patterns, and templates we are using in the Shopify App CLI tool. All the content is based on the CLI Design and Style Guidelines doc, as well as the guidance on the Shopify cli-ui Github repo.
+
+The most important principle to follow is **speed**. CLI tools are built to be extremely fast and perform tasks quickly. If there is ever a collision between heuristics, default to whatever results in the fastest perceived or actual performance. 
```

Comment:
I'm wondering if we should add somewhere in this about not using any gems or dependencies? I konw that's _technical_ design, but it just fits well with your speed principle

Reply:
I think this would fit better in Contributing.MD, this doc is very focused on the actual design not necessarily the technical reasons that help keep the tool fast.